src/server/json_family.cc

```diff
@@ -84,6 +85,42 @@ string JsonType(const json& val) {
   return "";
 }
 
+bool ParseDouble(string_view src, double* value) {
+  if (src.empty())
+    return false;
+
+  if (src == "-inf") {
+    *value = -HUGE_VAL;
+  } else if (src == "+inf") {
+    *value = HUGE_VAL;
+  } else {
+    absl::from_chars_result result = absl::from_chars(src.data(), src.end(), *value);
+    if (int(result.ec) != 0 || result.ptr != src.end() || isnan(*value))
+      return false;
+  }
+  return true;
+}
+
+template <typename T>
+void PrintOptVec(ConnectionContext* cntx, const OpResult<vector<optional<T>>>& result) {
+  if (result->empty()) {
+    (*cntx)->SendNullArray();
+  } else {
+    (*cntx)->StartArray(result->size());
+    for (auto& it : *result) {
+      if (it.has_value()) {
+        if constexpr (is_integral_v<T>) {
+          (*cntx)->SendLong(*it);
+        } else if (is_floating_point_v<T>) {
```

Comment:
nice!
1. I did not not that `constexpr else if` works.
2. still, I think it's better to do a regular `if` and then `static_assert` inside the block so you won't get a type in the future that it silently skips.